src/phoenix/datasets/experiments.py

```diff
@@ -170,3 +214,133 @@ async def async_run_experiment(example: ExampleProtocol) -> ExperimentPayload:
     return Experiment(
         id=experiment_id, dataset_id=dataset.id, dataset_version_id=dataset.version_id
     )
+
+
+def evaluate_experiment(
+    experiment: Experiment,
+    evaluator: ExperimentEvaluator,
+    name: Optional[str] = None,
+    label: Optional[str] = None,
+) -> None:
+    # define wrapper classes to coerce JSON payloads to conform to the input protocols until
+    # we flesh out our clientside models
+
+    @dataclass
+    class ExampleWrapper:
+        id: str
+        input: JSONSerializable
+        output: JSONSerializable
+
+    @dataclass
+    class RunWrapper:
+        id: str
+        output: JSONSerializable
+
+    client = _phoenix_client()
+
+    experiment_id = experiment.id
+    dataset_id = experiment.dataset_id
+    dataset_version_id = experiment.dataset_version_id
+
+    dataset_examples = client.get(
+        f"/v1/datasets/{dataset_id}/examples",
+        params={"version-id": str(dataset_version_id)},
+    ).json()
+
+    experiment_runs = client.get(f"/v1/experiments/{experiment_id}/runs").json()
+
+    # not all dataset examples have associated experiment runs, so we need to pair them up
+    # based on the example id, this assumes that the examples and experiment runs are sorted by id
+    example_run_pairs = []
+    example_cursor, run_cursor = 0, 0
```

Comment:
Suggestion: I think `index` is a simpler name. Cursor sounds like a graphQL cursor.

Reply:
I'm not sure there's a real difference between the two? I think they both indicate an object that tracks position pretty clearly